httpserver/http_connection.cpp

```diff
@@ -64,32 +64,40 @@ void make_http_request(boost::asio::io_context& ioc, const std::string& address,
                        uint16_t port, const std::shared_ptr<request_t>& req,
                        http_callback_t&& cb) {
 
-    error_code ec;
-    tcp::endpoint endpoint;
-    tcp::resolver resolver(ioc);
+    static tcp::resolver resolver(ioc);
 
-    tcp::resolver::iterator destination = resolver.resolve(address, "http", ec);
+    auto resolve_handler = [&ioc, address, port, req, cb = std::move(cb)](
+                               const boost::system::error_code& ec,
+                               boost::asio::ip::tcp::resolver::results_type
+                                   resolve_results) mutable {
+        if (ec) {
+            OXEN_LOG(error, "DNS resolution error for {}: {}", address,
+                     ec.message());
+            cb({SNodeError::ERROR_OTHER});
+            return;
+        }
 
-    if (ec) {
-        OXEN_LOG(error,
-                 "http: Failed to parse the IP address <{}>. Error code = {}. "
-                 "Message: {}",
-                 address, ec.value(), ec.message());
-        return;
-    }
-    while (destination != tcp::resolver::iterator()) {
-        const tcp::endpoint thisEndpoint = (destination++)->endpoint();
-        if (!thisEndpoint.address().is_v4()) {
-            continue;
+        tcp::endpoint endpoint;
+
+        while (resolve_results != tcp::resolver::iterator()) {
+            const tcp::endpoint ep = (resolve_results++)->endpoint();
+            if (!ep.address().is_v4()) {
+                continue;
+            }
+            endpoint = ep;
         }
-        endpoint = thisEndpoint;
-    }
-    endpoint.port(port);
+        endpoint.port(port);
```

Comment:
It looks like if I give this an address that doesn't have an IPv4 record we're going to get through the resolver loop without having set endpoint, but we aren't failing in that case.

Reply:
Now respond with an error